syft/pkg/license.go

```diff
@@ -29,6 +29,7 @@ type License struct {
 	Type           license.Type
 	URLs           []string         `hash:"ignore"`
 	Locations      file.LocationSet `hash:"ignore"`
+	Contents       string           `hash:"ignore"` // The optional binary contents of the license file
```

Comment:
If we add support for this we should make this configurable. It could be a simple on/off (collect or do not collect contents) but there is some argument for more options here. Specifically, when we can get the ID from contents and we have a high degree of confidence that it matches then including the contents is redundant / not necessary (one of multiple possible middle of the road options). 

I think the default for this option should either be "off" or the middle of the road option.

Reply:
My implementation **only** ever fills the ```Contents``` field if a **non-empty** file identified as potential license file (by name) exists and **no SPDX-expression** can be determined for it with the required confidence.
Otherwise the status quo implementation is retained.
Hence I would assume that the option as you describe it is currently "off" by default. (Unless of course in the event of a non-identifiable license.) 😉

I would advocate for a later implementation of the configurability.

Reply:
I would advocate for a later implementation of the configurability, as the current implementation mitigates the fact, that packages with an identified (but non-SPDX-compatible) license file will be erroneously handled as **unlicensed**.